client/my-sites/stats/stats-module/index.jsx

```diff
@@ -70,6 +93,75 @@ class StatsModule extends Component {
 			// eslint-disable-next-line react/no-did-update-set-state
 			this.setState( { loaded: false } );
 		}
+
+		const isRealTime = this.props.query?.interval !== undefined;
+		if ( ! isRealTime ) {
+			return;
+		}
+
+		// Limit data processing to match the query interval to avoid spurious updates.
+		// Need to convert from milliseconds to seconds for comparison.
+		const limit = this.props.query.interval / 1000;
+		const { lastUpdated } = this.state;
+		const now = moment();
+
+		if ( ! lastUpdated || now.diff( lastUpdated, 'seconds' ) >= limit ) {
+			const updatedHistory = this.updateHistory( this.state.dataHistory, this.props.data );
+			const firstSnapshot = updatedHistory[ 0 ];
+			const lastSnapshot = updatedHistory[ updatedHistory.length - 1 ];
+			const diffData = this.calculateDiff( firstSnapshot.data, lastSnapshot.data );
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState( {
+				diffData,
+				dataHistory: updatedHistory,
+				lastUpdated: now,
+			} );
+		}
+	}
+
+	updateHistory( history, data ) {
+		// Timestamp the new data snapshot.
+		const newSnapshot = {
+			timestamp: moment(),
+			data: data,
+		};
+
+		// Filter out snapshots older than minutesLimit prop.
+		// This determines the baseline for the diff calculation.
+		const { minutesLimit } = this.props;
+		const filteredHistory = [ ...history, newSnapshot ].filter(
+			( snapshot ) => moment().diff( snapshot.timestamp, 'minutes' ) <= minutesLimit
+		);
+
+		return this.compactHistory( filteredHistory );
+	}
+
+	compactHistory( history ) {
+		const MAX_HISTORY_LENGTH = 35;
+
+		if ( history.length > MAX_HISTORY_LENGTH ) {
+			// Keep every other entry to keep memory usage low.
+			return history.filter( ( _, index ) => index % 2 === 0 );
+		}
+
+		return history;
+	}
+
+	calculateDiff( prevData, newData ) {
+		// Create a lookup map for previous data using item IDs.
+		const prevDataMap = new Map( prevData.map( ( item ) => [ item.id, item ] ) );
+
+		// Calculate the difference value for each new item.
+		const diff = newData.map( ( item ) => {
+			// Pull matching data from previous snapshot, or default to 0 if not found.
+			const prevItem = prevDataMap.get( item.id ) || { value: 0 };
+			return {
+				...item,
+				diffValue: item.value - prevItem.value,
```

Comment:
Because we only compare the first and last data, the difference might be the exact views count when a certain item is just ordered to the top 10.

<img width="683" alt="截圖 2025-02-12 晚上11 53 56" src="https://github.com/user-attachments/assets/a9bbc181-aff0-4fec-bfde-0b77ed2c0488" />

Reply:
Good catch. I hadn't seen this in my testing but I understand why this is happening. Current value minus no previous value equals current value. I'll have another look at this as well.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/99940.